update lockfile for nrop-mg

abraham2512 opened the pull request:
This PR updates the lockfile in NROP MG, which includes a cve fix for rsync rpm

abraham2512 commented:
/ok-to-test

yanirq commented:
> Thanks @abraham2512 for working on this. I don't see a connection between the tekton updates and the lockfile updates, if you agree let's please split each update to separate PR, one for tekton and one for lockfile while considering the notes below:
> 
>     1. let's remove the disabled repos from the .repo file, it makes it slow to review the changes and takes time to load the file.
> 
>     2. please add a description to the commits touching on the problem that the commit solves and how, this is fine if it is brief.
> 
>     3. let's start the commit title with "konflux: " or "tekton: " based on the updated files

I agree with 1 and 3,   as far as for 2, adding description for the digest updates is not a must (since konflux PRs for that have pretty much the same commit message) .
@shajmakh  @abraham2512  I would actually keep the PR as a whole if it is the same content all the way down to 4.14  since this is still onboarding PRs and we want to stabilize the must gather d/s builds asap .

abraham2512 commented:
@shajmakh I agree with Yanir, the lockfile and tekton updates should not be split out, conformas will fail for separate reasons on both PRs. This will also double the number of backports that need to be created and approved, and I do not see a good enough reason to do this.
Other than this, as suggested I've added konflux, tekton to commit messages and removed the disabled repos.

shajmakh commented:
> @shajmakh I agree with Yanir, the lockfile and tekton updates should not be split out, conformas will fail for separate reasons on both PRs. This will also double the number of backports that need to be created and approved, and I do not see a good enough reason to do this.
> Other than this, as suggested I've added konflux, tekton to commit messages and removed the disabled repos.

Thanks. Usually it is always better to have separate PRs for separate problems but I get the urgency of this and the concern about the backports and given that this is still considered onboarding, multiple commits is good enough. 
Please squash the last commit into the first one and add a description of what the problem was with the current rpms (likely low CHI score).

abraham2512 commented:
Squashed commits 1 and 4 and added a description @shajmakh

yanirq commented:
/lgtm
/approve

shajmakh commented:
/hold

shajmakh commented on .tekton/build-pipeline-must-gather.yaml:
```diff
@@ -98,6 +98,10 @@ spec:
       set of values is determined by the configuration of the multi-platform-controller.
     name: build-platforms
     type: array
+  - default: "false"
```
The commit contains disabling and enabling two different checks, please align the commit title and rephrase the description, and remove double sign-off

abraham2512 commented on .konflux/must-gather/redhat.repo:
```diff
@@ -0,0 +1,52 @@
+#
```
Done

abraham2512 commented on .konflux/must-gather/redhat.repo:
```diff
@@ -0,0 +1,52 @@
+#
+
+[codeready-builder-for-rhel-9-$basearch-rpms]
+name = Red Hat CodeReady Linux Builder for RHEL 9 $basearch (RPMs)
+baseurl = https://cdn.redhat.com/content/dist/rhel9/9.4/$basearch/codeready-builder/os
```
We cannot have both baseos-rpms and baseos-eus-rpms, because only baseos-eus-rpms has the rsync CVE fix

abraham2512 replied:
Removed appstream repo

shajmakh commented on .konflux/must-gather/redhat.repo:
```diff
@@ -0,0 +1,52 @@
+
+[rhel-9-for-$basearch-baseos-eus-rpms]
+name = Red Hat Enterprise Linux 9 for $basearch - BaseOS - Extended Update Support (RPMs)
+baseurl = https://cdn.redhat.com/content/eus/rhel9/9.4/$basearch/baseos/os
```
> What do we gain by condensing it into this file?


In case of an update in rhel version, if the repos are embedded within the input file, you will not need to update the corresponding rhel versions because it would fetch the corresponding RPMs based on the rhel version from the final base image that is used in the dockerfile via the {version} parameter. What need to be modified is only the entitlement IDs, this saves us an extra step in "generating" the repo file. 

Both approaches work, and I see different tastes here; given that this is not forming any blocker we can go with this update, but please state in the same commit message reflecting why you chose to switch to this approach instead of updating the repos inplace within the input file (e.g following the pattern used in the other container repos.)

abraham2512 commented:
/retest

shajmakh commented:
/approve
/lgtm
Thanks for addressing the comments!
